config/routes.rb

```diff
@@ -301,6 +301,7 @@
   get "session/current" => "session#current"
   get "session/csrf" => "session#csrf"
   get "composer_messages" => "composer_messages#index"
+  post "composer/parse_html" => "composer#parse_html"
```

Comment:
A controller test or 2 please! 

What if the html content is 3 megabytes? can this take out the parser?

Reply:
we already having spec for this [here](https://github.com/discourse/discourse/blob/master/spec/components/html_to_markdown_spec.rb). Should I check this against large input data?

Reply:
is there any js test about this somewhere? couldn’t see it?
At least one simple test would be cool, fillIn("xxx", textearea), check conversion in md and html in preview, should be easy enough to do.

Reply:
Ruby spec added. @jjaffeux for Qunit test I have to trigger paste event manually and also have to wait for the server response to render the response in editor. Is there any possibility for that?

Reply:
Do you ? I don’t think pasting triggers a request.

Reply:
no in this particular functionality I am calling web request whenever paste event triggered with html data.